src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java

```diff
@@ -1352,6 +1353,51 @@ public ItemStack get(PlayerArmorChangeEvent e) {
 				}
 			}, 0);
 		}
+		//PlayerInventorySlotChangeEvent
+		if (Skript.classExists("io.papermc.paper.event.player.PlayerInventorySlotChangeEvent")) {
+			EventValues.registerEventValue(PlayerInventorySlotChangeEvent.class, ItemStack.class, new Getter<ItemStack, PlayerInventorySlotChangeEvent>() {
+				@Override
+				@Nullable
+				public ItemStack get(PlayerInventorySlotChangeEvent event) {
+					return event.getNewItemStack();
+				}
+			}, EventValues.TIME_NOW);
+			EventValues.registerEventValue(PlayerInventorySlotChangeEvent.class, ItemStack.class, new Getter<ItemStack, PlayerInventorySlotChangeEvent>() {
+				@Override
+				@Nullable
+				public ItemStack get(PlayerInventorySlotChangeEvent event) {
+					return event.getOldItemStack();
+				}
+			}, EventValues.TIME_PAST);
+			EventValues.registerEventValue(PlayerInventorySlotChangeEvent.class, ItemStack.class, new Getter<ItemStack, PlayerInventorySlotChangeEvent>() {
+				@Override
+				@Nullable
+				public ItemStack get(PlayerInventorySlotChangeEvent event) {
+					return event.getNewItemStack();
+				}
+			}, EventValues.TIME_FUTURE);
+			EventValues.registerEventValue(PlayerInventorySlotChangeEvent.class, Slot.class, new Getter<Slot, PlayerInventorySlotChangeEvent>() {
+				@Override
+				@Nullable
+				public Slot get(PlayerInventorySlotChangeEvent event) {
+					PlayerInventory inv = event.getPlayer().getInventory();
+					int slotIndex = event.getSlot();
+					// Not all indices point to inventory slots. Equipment, for example
+					if (slotIndex >= 36) {
+						return new ch.njol.skript.util.slot.EquipmentSlot(event.getPlayer(), slotIndex);
+					} else {
+						return new InventorySlot(inv, slotIndex);
+					}
+				}
+			}, EventValues.TIME_NOW);
+			EventValues.registerEventValue(PlayerInventorySlotChangeEvent.class, Number.class, new Getter<Number, PlayerInventorySlotChangeEvent>() {
+				@Override
+				@Nullable
+				public Number get(PlayerInventorySlotChangeEvent event) {
+					return event.getSlot();
+				}
+			}, EventValues.TIME_NOW);
```

Comment:
I think this is not needed, as we can use `index of event-slot`

Reply:
Good point, I had added it because I wasn't sure if I was going to keep event-slot, but you're right that only one of the two is necessary.
I'll remove it if/when we decide to keep event-slot.

Reply:
event-slot is good, it's used in other places so keep it

Reply:
Sounds good! While you're here, do you have any opinions on whether `event-item` should be the new or the old itemstack?

Reply:
`event-item` should be the future one even `future event-item`, only `past event-item` should be the old one